plugins/migration/bin/fix-imports.js

```diff
@@ -0,0 +1,246 @@
+
+import {createSubCommand, extractRequiredInfo} from './command.js';
+import {readFileSync, statSync, writeFileSync} from 'fs';
+
+export const fixImports = createSubCommand(
+  'fix-imports',
+  '>=9',
+  'Add imports for modules that have been moved out of the Blockly namespace')
+  .option(
+      '-i, --in-place [suffix]',
+      'fix imports in-place, optionally create backup files with the ' +
+      'given suffix. Otherwise output to stdout')
+  .action(function() {
+    // TODO (#1211): In the future we should use the fromVersion and toVersion
+    //   so that we can support doing this across multiple versions. But for
+    //   now we just need it to work for v9.
+    const {fileNames} = extractRequiredInfo(this);
+
+    fileNames.forEach((name) => {
+      if (statSync(name).isDirectory()) return;
+      const contents = readFileSync(name, 'utf8');
+      const migratedContents = migrateContents(contents);
+      const inPlace = this.opts().inPlace;
+
+
+      if (inPlace) {
+        if (typeof inPlace == 'string') {
+          writeFileSync(name + inPlace, contents);
+        }
+        writeFileSync(name, migratedContents);
+        process.stderr.write(`Migrated renamings in ${name}\n`);
+      } else {
+        process.stdout.write(migratedContents + '\n');
+      }
+    });
+  });
+
+/**
+ * @typedef {{
+ *   import: string,
+ *   oldIdentifier?: string,
+ *   newIdentifier: string,
+ *   newImport: string,
+ *   newRequire: string,
+ * }}
+ */
+let MigrationData;
+
+// TODO (#1211): Make this database format more robust.
+/** @type {MigrationData[]} */
+const database = [
+  {
+    import: 'blockly/dart',
+    newIdentifier: 'dartGenerator',
+    newImport: `import {dartGenerator} from 'blockly/dart';`,
+    newRequire: `const {dartGenerator} = require('blockly/dart');`
+  },
+  {
+    import: 'blockly/javascript',
+    oldIdentifier: 'Blockly.JavaScript',
+    newIdentifier: 'javascriptGenerator',
+    newImport: `import {javascriptGenerator} from 'blockly/javascript';`,
+    newRequire: `const {javascriptGenerator} = require('blockly/javascript');`
+  },
+  {
+    import: 'blockly/lua',
+    newIdentifier: 'luaGenerator',
+    newImport: `import {luaGenerator} from 'blockly/lua';`,
+    newRequire: `const {luaGenerator} = require('blockly/lua');`
+  },
+  {
+    import: 'blockly/php',
+    newIdentifier: 'phpGenerator',
+    newImport: `import {phpGenerator} from 'blockly/php';`,
+    newRequire: `const {phpGenerator} = require('blockly/php');`
+  },
+  {
+    import: 'blockly/python',
+    newIdentifier: 'pythonGenerator',
+    newImport: `import {pythonGenerator} from 'blockly/python';`,
+    newRequire: `const {pythonGenerator} = require('blockly/python');`
+  },
+  {
+    import: 'blockly/blocks',
+    oldIdentifier: 'Blockly.libraryBlocks',
+    newIdentifier: 'libraryBlocks',
+    newImport:  `import {libraryBlocks} from 'blockly/blocks';`,
+    newRequire: `const {libraryBlocks} = require('blockly/blocks');`
```

Comment:
I think you may have gotten the wrong end of the stick about how the library blocks are exported.  This will be mostly irrelevant as until fairly recently we only exported the actual blocks dictionary and AFAIK most developers only imported `blockly/blocks` for side effects, but I believe this section should be as follows:

```suggestion
    import: 'blockly/blocks',
    oldIdentifier: 'Blockly.libraryBlocks',
    newIdentifier: 'libraryBlocks',
    newImport:  `import * as libraryBlocks from 'blockly/blocks';`,
    newRequire: `const libraryBlocks = require('blockly/blocks');`
```